sound/soc/sof/intel/hda.h

```diff
@@ -685,45 +685,15 @@ void hda_sdw_process_wakeen(struct snd_sof_dev *sdev);
 
 #else
 
-static inline int hda_sdw_acpi_scan(struct snd_sof_dev *sdev)
-{
-	return 0;
-}
-
-static inline int hda_sdw_probe(struct snd_sof_dev *sdev)
-{
-	return 0;
-}
-
 static inline int hda_sdw_startup(struct snd_sof_dev *sdev)
 {
 	return 0;
 }
 
-static inline int hda_sdw_exit(struct snd_sof_dev *sdev)
-{
-	return 0;
-}
-
 static inline void hda_sdw_int_enable(struct snd_sof_dev *sdev, bool enable)
 {
 }
 
-static inline bool hda_dsp_check_sdw_irq(struct snd_sof_dev *sdev)
-{
-	return false;
-}
-
-static inline irqreturn_t hda_dsp_sdw_thread(int irq, void *context)
-{
-	return IRQ_HANDLED;
-}
-
-static inline bool hda_sdw_check_wakeen_irq(struct snd_sof_dev *sdev)
-{
-	return false;
-}
-
```

Comment:
I have mixed feelings about this one. The intent was to keep all SoundWire stuff in one place so that it was self-explanatory. There are e.g. 3 phases for scan, probe and startup and that logic is now broken up.

Reply:
These functions are local to hda.c, they are not used outside and the implementations are within #if IS_ENABLED()
The hda.h is included to all HDA related files and it is a bit confusing to have inline prototypes which should be local to hda.c.

Reply:
ah yes, now I see what you meant, it's already the case that in the IS_ENABLED case we have 3 functions only. I forget we made the other static already.